### PR TITLE
New ad network: sunmedia

### PR DESCRIPTION
**Pull request by operezes**
Hi Team 
We are interested to integrate our ad plugin (sunmedia) in AMP

——
Example:
http://static.addevweb.com/SMVpaidCreatives/amp/examples/ads.amp.max.html

* You have to scroll until you watch the ad in the middle of text block
——

I have signed CLA doc as operezes@yahoo.es, and in GitHub  I am using same email account.

If you need any additional information or whatever from my side feel free to ask me 

Best regards
Oscar

Closes https://github.com/ampproject/amphtml/issues/7426.

### CODE REVIEW

**Comment by lannka** [comment]
@operezes would you follow the [guide](https://github.com/ampproject/amphtml/blob/master/ads/README.md#developer-guidelines-for-a-pull-request) and implement render-start and no-content APIs?

**Comment by operezes** [comment]
Hi Iannka, nice to meet you

I had implemented render-start inside my js plugin, which is loaded after AMP loads main sunmedia.js. no-content was not implemented, but I already have added it to that js:

https://vod.addevweb.com/sunmedia/amp/ads/SMIntextAMP.js

render-sart - line 465
no-content: line 199

Thank you very much
 Oscar

**Comment by lannka** [comment]
@operezes in that case, you will need add `renderStartImplemented: true,` to the config. otherwise the events are ignored.

**Comment by operezes** [comment]
Hi Iannka, I already have added that parameter (renderStartImplemented: true) in config.js file

Best regards
Oscar

**Comment by lannka** [comment]
Please add your example to `ads.amp.html`. Also pls update `amp-ad.md`

**Comment by operezes** [comment]
Hi Iannka, I already have updated ads.amp.html and amp-ad.md

Thank you very much
Oscar

**Comment by lannka** [comment]
@operezes did you make your example work locally?
I got a console error:
> Uncaught TypeError: Cannot read property 'cid' of undefined
>    at SMIntext.init (SMIntextAMP.js:16)
>   at window.smintextstart (SMIntextAMP.js:510)
>   at HTMLVideoElement.<anonymous> (SMIntextAMP.js:498)

Check [here](https://github.com/ampproject/amphtml/blob/master/ads/README.md#developer-guidelines-for-a-pull-request) if you don't know how to run it locally.

**Comment by operezes** [comment]
Hi Iannka

Sorry for the mess here, we had created other fully local testing environment.
We have updated integration following all instructions described in "amphtml" doc.

We have updated:

/ads/sunmedia.js
/ads/sunmedia.md
/ads/_config.js
/3p/integration.js
/extensions/amp-ad/amp-ad.md
/examples/ads.amp.html

You can check testing integration here, running gulp server:
http://localhost:8000/examples/ads.amp.max.html?type=sunmedia

We have forced one ad for that host

Thank you very much and sorry for the confusion
Oscar

File: 3p/frame.max.html [hunk 1]
```diff
@@ -6,7 +6,17 @@
 </head>
 <body style="margin:0">
 <div id="c" style="position:absolute;top:0;left:0;bottom:0;right:0;">
-  <script>draw3p()</script>
+  <script>
```
Comment (lannka) [inline 3p/frame.max.html]: can you revert changes here?

File: 3p/remote.html [hunk 2]
```diff
@@ -4,23 +4,25 @@
 <meta name="robots" content="noindex">
 <script>
 (function() {
-var v = location.search.substr(1);
-if (!(/^\d+(-canary)?$/.test(v))) return;
-var u = 'https://3p.ampproject.net/'+encodeURIComponent(v)+'/f.js';
-document.write('<script'+' src="'+encodeURI(u)+'"><'+'/script>');
+//var v = location.search.substr(1);
```
Comment (lannka) [inline 3p/remote.html]: here as well

File: examples/ads.amp.html [hunk 3]
```diff
@@ -1158,6 +1159,17 @@ <h2 id="sovrn">SOVRN</h2>
     <div fallback></div>
   </amp-ad>
 
+  <h2 id="sunmedia">SunMedia</h2>
+  <amp-ad width="300" height="1"
+      type="sunmedia"
+      layout="responsive"
+      data-cskp=1
```
Comment (lannka) [inline examples/ads.amp.html]: double quote the value

File: examples/ads.amp.html [hunk 4]
```diff
@@ -1158,6 +1159,17 @@ <h2 id="sovrn">SOVRN</h2>
...
+      data-cid="sunmedia_test"
+      data-crst=1>
```
Comment (lannka) [inline examples/ads.amp.html]: same here

File: 3p/frame.max.html (hunk 1 above)
Comment (lannka) [inline 3p/frame.max.html]: please revert the file

File: 3p/remote.html [hunk 5]
```diff
@@ -14,13 +14,14 @@
 <body style="margin:0">
 <div id="c" style="position:absolute;top:0;left:0;bottom:0;right:0;">
   <script>
-    draw3p(undefined,
+    /*draw3p(undefined,
```
Comment (lannka) [inline 3p/remote.html]: revert the file

**Comment by operezes** [comment]
Hi Iannka, sorry for the confusion here, I already have left the files as  original project

File: 3p/frame.max.html [hunk 6]
```diff
@@ -10,4 +10,4 @@
 </div>
 <script>if (window.docEndCallback) window.docEndCallback()</script>
 </body>
-</html>
+</html>
```
Comment (lannka) [inline 3p/frame.max.html]: please revert

File: 3p/remote.html [hunk 7]
```diff
@@ -25,4 +25,4 @@
 </div>
 <script>if (window.docEndCallback) window.docEndCallback()</script>
 </body>
-</html>
+</html>
```
Comment (lannka) [inline 3p/remote.html]: please revert

File: ads/sunmedia.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {loadScript, validateData} from '../3p/3p';
+/**
```
Comment (lannka) [inline ads/sunmedia.js]: nit: add a blank line below import

File: ads/sunmedia.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    allowed_data: ['cskp', 'crst', 'cdb', 'cid'],
+    mandatory_data: ['cid'],
+    data,
+  };
+
+  validateData(data,
+    global._sunmedia_amp.mandatory_data, global._sunmedia_amp.allowed_data);
+
+  loadScript(global, 'https://vod.addevweb.com/sunmedia/amp/ads/SMIntextAMP.js');
+
```
Comment (lannka) [inline ads/sunmedia.js]: nit: remove blank line here

File: examples/ads.amp.html [hunk 10]
```diff
@@ -1158,6 +1159,17 @@ <h2 id="sovrn">SOVRN</h2>
...
   </amp-ad>
 
+  <h2 id="sunmedia">SunMedia</h2>
+  <amp-ad width="300" height="1"
+      type="sunmedia"
+      layout="responsive"
+      data-cskp="1"
+      data-cid="sunmedia_test"
+      data-crst="1">
+    <div placeholder></div>
```
Comment (lannka) [inline examples/ads.amp.html]: please sync to the latest code and remove the placeholder and fallback divs

**Comment by lannka** [comment]
please also fix the Travis build

File: examples/ads.amp.html [hunk 11]
```diff
@@ -1158,6 +1159,15 @@ <h2 id="sovrn">SOVRN</h2>
     <div fallback></div>
   </amp-ad>
 
+  <h2 id="sunmedia">SunMedia</h2>
+  <amp-ad width="300" height="1"
```
Comment (lannka) [inline examples/ads.amp.html]: please change height to a bigger value to avoid a resize. for e.g.: `250`

**Comment by operezes** [comment]
Hi Iannka,

I have watched you have merged our plugin  and  I can see the plugin included in the project, but I am trying to check the plugin in a real test page, and I get the next error message: 

Type "sunmedia" is not supported in amp-ad

Should I do anything more to be approved by AMP to be able using sunmedia.js in live environment?

Thank you for your help
Oscar

**Comment by lannka** [comment]
@operezes you will need to wait a release cycle to be able to test it on live environment, which should be next Thursday.

**Comment by operezes** [comment]
Hi Iannka, ok, thank you very much for your help :-)
Oscar

**Pull request merged**